### PR TITLE
[amp-sidebar 1.0] Un-version sidebar 1.0

### LINKED ISSUE #10770: when open side bar, click in menu item to scroll to an element and them close side bar the page scroll up
## What's the issue?

when open side bar, click in menu item to scroll to an element and them close side bar the page scroll up

## How do we reproduce the issue?

here [Demo](https://mauricionr.github.io/demos/issues/10770/)

1. Open side bar
2. go to language
3. close side bar
## What browsers are affected?

i guess all

## Which AMP version is affected?

i guess all versions

### PR DESCRIPTION
**Pull request by torch2424**
closes #10770

By request of the validation team, and approval by @camelburrito and @dvoytenko , this PR will unversion sidebar.

`toolbar` will now live as a feature to `amp-sidebar`, and is hidden under the experimental flag, `amp-sidebar toolbar`. This PR does the following changes:

* Moves all approved code and tests from 1.0 into 0.1
* Adds an example for #10770 (By request of @camelburrito to be done in the same PR)
* Updates the test manual for amp-sidebar toolbar
* Removes amp-sidebar 1.0 from the gulp file.
* Hides toolbar features under the experimental flag

**History of toolbar.js to ensure it has not changed since PR was opened:** https://github.com/ampproject/amphtml/commits/master/extensions/amp-sidebar/1.0/toolbar.js

See the screenshots below showing it will not affect current users, but only experimental toolbar users:

screen shot 2017-08-03 at 6 33 00 pm
screen shot 2017-08-03 at 6 32 47 pm
screen shot 2017-08-03 at 6 32 08 pm
screen shot 2017-08-03 at 6 31 53 pm
screen shot 2017-08-03 at 6 31 46 pm

### CODE REVIEW

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 1]
```diff
@@ -15,13 +15,19 @@
  */
 
 import {CSS} from '../../../build/amp-sidebar-0.1.css';
+import {Layout} from '../../../src/layout';
+import {dev, user} from '../../../src/log';
+import {isExperimentOn} from '../../../src/experiments';
 import {KeyCodes} from '../../../src/utils/key-codes';
```
Comment (camelburrito) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Move K between C & L

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 2]
```diff
@@ -91,6 +101,25 @@ export class AmpSidebar extends AMP.BaseElement {
       this.element.setAttribute('side', this.side_);
     }
 
+    const ampdoc = this.getAmpDoc();
+    // Get the toolbar attribute from the child navs
+    const toolbarElements =
+      toArray(this.element.querySelectorAll('nav[toolbar]'));
+
+    if (toolbarElements.length > 0) {
+      user().assert(isExperimentOn(this.win, TAG),
```
Comment (camelburrito) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: move the experiment check above - don't assert

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 3]
```diff
@@ -154,6 +184,27 @@ export class AmpSidebar extends AMP.BaseElement {
...
 
+  /** @override */
+  activate() {
+    this.open_();
+  }
+
+  /** @override */
+  onLayoutMeasure() {
+    // Check our toolbars for changes
+    this.toolbars_.forEach(toolbar => {
```
Comment (camelburrito) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: experiment guard.

File: extensions/amp-sidebar/amp-sidebar.md [hunk 4]
```diff
@@ -53,21 +61,42 @@ limitations under the License.
...
+    <li>Nav item 3</li>
+    <li>Nav item 4</li>
+    <li>Nav item 5</li>
+    <li>Nav item 6</li>
   </ul>
 </amp-sidebar>
 ```
 
+*Example: Advanced usage of `amp-sidebar` with anchor links and smooth scrolling*
+
```
Comment (camelburrito) [inline extensions/amp-sidebar/amp-sidebar.md]: This is not advanced usage - make this a part of the default example

File: extensions/amp-sidebar/0.1/test/test-amp-sidebar.js [hunk 5]
```diff
@@ -54,6 +55,52 @@ describes.realWin('amp-sidebar 0.1 version', {
         const anchor = iframe.doc.createElement('a');
         anchor.href = '#section1';
         ampSidebar.appendChild(anchor);
+        if (options.toolbars) {
```
Comment (camelburrito) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: create a separate describe for tests with toolbar

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 1 above)
Comment (torch2424) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Done

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 2 above)
Comment (torch2424) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: done

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 3 above)
Comment (torch2424) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Done

File: extensions/amp-sidebar/amp-sidebar.md (hunk 4 above)
Comment (torch2424) [inline extensions/amp-sidebar/amp-sidebar.md]: Done

File: extensions/amp-sidebar/0.1/test/test-amp-sidebar.js (hunk 5 above)
Comment (torch2424) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: Done

File: extensions/amp-sidebar/0.1/test/test-amp-sidebar.js [hunk 6]
```diff
@@ -35,55 +36,95 @@ describes.realWin('amp-sidebar 0.1 version', {
...
+            'mutate', callback => {
+              callback();
+            });
+        sandbox.stub(impl.vsync_,
+            'mutatePromise', callback => {
+              callback();
+              return Promise.resolve();
+            });
+         // Create our individual toolbars
+        options.toolbars.forEach(toolbarObj => {
```
Comment (camelburrito) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: move this to its own function

File: extensions/amp-sidebar/amp-sidebar.md [hunk 7]
```diff
@@ -104,6 +117,89 @@ Example:
 <button on='tap:sidebar1.close'>x</button>
 ```
 
+### Toolbar
```
Comment (camelburrito) [inline extensions/amp-sidebar/amp-sidebar.md]: mark it as experimental (Experimental)

File: extensions/amp-sidebar/0.1/test/test-amp-sidebar.js (hunk 6 above)
Comment (torch2424) [inline extensions/amp-sidebar/0.1/test/test-amp-sidebar.js]: Done

File: extensions/amp-sidebar/amp-sidebar.md (hunk 7 above)
Comment (torch2424) [inline extensions/amp-sidebar/amp-sidebar.md]: Done

**Pull request merged**